multistep/commonsteps/http_config.go

```diff
@@ -23,6 +23,15 @@ type HTTPConfig struct {
 	// started. The address and port of the HTTP server will be available as
 	// variables in `boot_command`. This is covered in more detail below.
 	HTTPDir string `mapstructure:"http_directory"`
+	// Key/Values to serve using an HTTP server. http_content works like and
+	// conflicts with http_directory The keys represent the paths and the values
+	// contents. This is useful for hosting kickstart files and so on. By
+	// default this is empty, which means no HTTP server will be started. The
+	// address and port of the HTTP server will be available as variables in
+	// `boot_command`. This is covered in more detail below. Example: Setting
+	// `"foo/bar"="baz", will allow you to http get on
+	// `http://{http_ip}:{http_port}/foo/bar`.
+	HTTPContent map[string]string `mapstructure:"http_content"`
```

Comment:
This is not a backward incompatibility but it is something that will require all builder components using http_config to update their hcl2spec files in order to be compatible with this feature.
If I'm right, I have two questions: 
- Are you planning to write a blog post for this?  
- Should this be officially released on 1.8.0?

Reply:
Good questions!
> Should this be officially released on 1.8.0?

The earlier we merge this one, the less manual updates we will have to do in plugins in order to support this. So my guess would be to merge this one now-ish.


> Are you planning to write a blog post for this?

🤔  I wasn't planning on doing that ! But you are right !
This definitely could use a new guide or help section too.

Reply:
Nice! We just need to remember to run generate on Packer when this one is merged. So that 1.7.1 will be compatible with these changes.